rust/operator-binary/src/druid_connection_controller.rs

```diff
@@ -0,0 +1,316 @@
+    },
+    #[snafu(display("object is missing metadata to build owner reference"))]
+    ObjectMissingMetadataForOwnerRef {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display(
+        "Failed to get Druid connection string from config map {} in namespace {:?}",
+        cm_name,
+        namespace
+    ))]
+    GetDruidConnStringConfigMap {
+        source: stackable_operator::error::Error,
+        cm_name: String,
+        namespace: Option<String>,
+    },
+    #[snafu(display(
+        "Failed to get Druid connection string from config map {} in namespace {:?}",
+        cm_name,
+        namespace
+    ))]
+    MissingDruidConnString {
+        cm_name: String,
+        namespace: Option<String>,
+    },
+    #[snafu(display(
+        "druid connection state is 'importing' but failed to find job {}/{}",
+        namespace,
+        name
+    ))]
+    GetImportJob {
+        source: stackable_operator::error::Error,
+        namespace: String,
+        name: String,
+    },
+    #[snafu(display("Failed to check if druid discovery map exists"))]
+    DruidDiscoveryCheck {
+        source: stackable_operator::error::Error,
+    },
+    SupersetDBExistsCheck {
+        source: stackable_operator::error::Error,
+    },
+    SupersetDBRetrieval {
+        source: stackable_operator::error::Error,
+    },
+}
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub async fn reconcile_druid_connection(
+    druid_connection: DruidConnection,
+    ctx: Context<Ctx>,
+) -> Result<ReconcilerAction> {
+    tracing::info!("Starting reconciling DruidConnections");
+
+    let client = &ctx.get_ref().client;
+
+    if let Some(ref s) = druid_connection.status {
+        match s.condition {
+            DruidConnectionStatusCondition::Provisioned => {
+                // Is the superset DB object there, and is its status "Ready"?
+                let superset_db_ready = if client
+                    .exists::<SupersetDB>(
+                        &druid_connection.spec.superset_cluster_name,
+                        Some(&druid_connection.spec.superset_cluster_namespace),
+                    )
+                    .await
+                    .context(SupersetDBExistsCheck)?
+                {
+                    let superset_db_status = client
+                        .get::<SupersetDB>(
+                            &druid_connection.spec.superset_cluster_name,
+                            Some(&druid_connection.spec.superset_cluster_namespace),
+                        )
+                        .await
+                        .context(SupersetDBRetrieval)?
+                        .status;
+                    if let Some(s) = superset_db_status {
+                        s.condition == SupersetDBStatusCondition::Ready
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                };
```

Comment:
This is a little messy. You get an error (and reconcile) if it does not exist. You have the same behavior from just using get? Maybe something like that
``` 
let mut superset_db_ready = false;
if let Some(status) =
client
    .get::<SupersetDB>(
        &druid_connection.spec.superset_cluster_name,
        Some(&druid_connection.spec.superset_cluster_namespace),
    )
    .await
    .context(SupersetDBRetrieval)?.status {
    superset_db_ready = status.condition == SupersetDBStatusCondition::Ready;
}
```

Reply:
much better